### PR TITLE
Add O(1) `contains()` implementation for Range with Comparable Element

### PR DESCRIPTION
**Pull request by an0**

### CODE REVIEW

File: stdlib/public/core/Range.swift [hunk 1]
```diff
@@ -147,6 +147,29 @@ public struct Range<
...
+extension Range where Element: Comparable {
+  @warn_unused_result
+  func _customContainsEquatableElement(element: Generator.Element) -> Bool? {
+    return element >= self.startIndex && element < self.endIndex
+  }
+    
+  // FIXME: copied from SequenceAlgorithms as a workaround for https://bugs.swift.org/browse/SR-435
+  @warn_unused_result
+  public func contains(element: Generator.Element) -> Bool {
+    if let result = _customContainsEquatableElement(element) {
```
Comment (kballard) [inline stdlib/public/core/Range.swift]: This condition can never fail, so you may as well replace the whole body of this method with the same `return element >= self.startIndex && element < self.endIndex` that you use in `_customContainsEquatableElement`.

Comment (an0) [inline stdlib/public/core/Range.swift]: As stated in comment, it is a temp workaround. The whole `contains` function will be deleted after the bug is fixed.

**Comment by gribozavr** [comment]
@an0 Sorry for the delayed response!  Could you resolve conflicts against the `master` branch?  We split the `StdlibUnittest` library into two, `StdlibUnittest` and `StdlibCollectionUnittest`.

The patch LGTM, could you resolve conflicts and I'll merge?

**Comment by an0** [comment]
@gribozavr I just merged and resolved the conflicts. Can you check please?

**Comment by gribozavr** [comment]
@an0 Thank you for working on this!  Sorry, I don't think that the merge of `StdlibUnittest.swift.gyb` was successful -- all of the indentation is gone.  Could you try again, squashing your changes?

```diff
diff --git a/stdlib/private/StdlibUnittest/StdlibUnittest.swift.gyb b/stdlib/private/StdlibUnittest/StdlibUnittest.swift.gyb
index 06a3852..a51625c 100644
--- a/stdlib/private/StdlibUnittest/StdlibUnittest.swift.gyb
+++ b/stdlib/private/StdlibUnittest/StdlibUnittest.swift.gyb
@@ -25,90 +25,90 @@ import ObjectiveC
 #endif

 public struct SourceLoc {
-public let file: String
-public let line: UInt
-public let comment: String?
+  public let file: String
+  public let line: UInt
+  public let comment: String?

-public init(_ file: String, _ line: UInt, comment: String? = nil) {
-self.file = file
-self.line = line
-self.comment = comment
-}
+  public init(_ file: String, _ line: UInt, comment: String? = nil) {
+    self.file = file
+    self.line = line
+    self.comment = comment
+  }
```

**Comment by an0** [comment]
@gribozavr all the whitespace should be back now.

**Comment by gribozavr** [comment]
Thanks!

**Comment by gribozavr** [comment]
@swift-ci Please test OS X platform

**Comment by gribozavr** [comment]
@swift-ci Please test OS X platform

File: validation-test/stdlib/SequenceType.swift.gyb [hunk 2]
```diff
@@ -608,6 +608,21 @@ SequenceTypeTests.test("Set<T>.contains/CustomImplementation/${dispatch}") {
 
 % end
 
+SequenceTypeTests.test("Range<Element>.contains/WhereElementIsComparable/dispatch") {
+  MinimalComparableIndexValue.timesLessWasCalled = 0
```
Comment (gribozavr) [inline validation-test/stdlib/SequenceType.swift.gyb]: Sorry, this failed with:

```
swift/validation-test/stdlib/SequenceType.swift.gyb:612:52: error: cannot assign value of type 'Int' to type 'ResettableValue<Int>'
  MinimalComparableIndexValue.timesLessWasCalled = 0
```

**Comment by gribozavr** [comment]
@an0 Sorry, CI reported a failure (pasted above), could you take a look?

**Comment by an0** [comment]
@gribozavr I'm not sure how to check whether this specific test succeeds or not using `utils/build-script -t`. But I manually ran the test code in swift repl and it passed.

**Comment by modocache** [comment]
@an0 The test failure is part of the validation test suite: `swift/validation-test/stdlib/SequenceType.swift.gyb:612:52`.

You can run the entire suite using `built-script -T` (capital `T`).

Or you can run the individual test using by invoking `lit` (the test runner directly). Make sure that Swift and LLVM are built, then from your `swift` directory execute:

```
apple/swift $ ../llvm/utils/lit/lit.py -v ../build/Ninja-ReleaseAssert/swift-macosx-x86_64/validation-test-macosx-x86_64/stdlib/SequenceType.swift.gyb
```

(Note that this assumes you built in release mode with assertions--`Ninja-ReleaseAssert`--and that you are building on OS X--`swift-macosx-x86_64/validation-test-macosx-x86_64`.)

Hope that helps!

**Comment by an0** [comment]
@modocache thanks. I've verified that the test passed this time. cc @gribozavr

**Comment by an0** [comment]
@gribozavr can you check and merge it if there are no issues. I worry that there will be new conflicts again if delayed — I've already resolved conflicts twice. Thanks.

**Comment by gribozavr** [comment]
@an0 Thank you!

**Pull request merged**